summit/roiGeneration.m

```diff
@@ -1,340 +1,398 @@
+        % on the top-most well
+        title('Please zoom in on the the top-most row of the array and press any key.')
+        
+        % use mouse button to zoom in or out
+        zoom on;   
+        pause()
+        zoom off;
+    
+        % prompt user to click on the center of the top-most well
+        title('Please click on the center of a well in top row of the array.');
+        
+        [x_click,y_click] = ginput(1);
+        plot(x_click(1),y_click(1),'r+','MarkerSize',10);
+        % store the user-selected coordinates of the lower right well
+        x_topwell = x_click;
+        y_topwell = y_click;
+        
+        zoom out
+        array_bounds(3,:) = [x_topwell, y_topwell];
+        
+        
+        
+        % Change display in imaged window to indicate user should zoom in
+        % on the bottom-most well
+        title('Please zoom in on the the bottom-most row of the array (spaced at least one column from the edge of the image) and press any key.')
+        
+        % use mouse button to zoom in or out
+        zoom on;   
+        pause()
+        zoom off;
+    
+        % prompt user to click on the center of the bottom-most well
+        title('Please click on the center of a well in the bottom-most row of the array.');
+        
+        [x_click,y_click] = ginput(1);
+        plot(x_click(1),y_click(1),'r+','MarkerSize',10);
+        
+        % store the user-selected coordinates of the lower right well
+        x_bottomwell = x_click;
+        y_bottomwell = y_click;
+        
+        zoom out
+        array_bounds(4,:) = [x_bottomwell, y_bottomwell];
+        
+        % store all of the coordinates of the array bounds to the struct
+        struct.array_bounds = array_bounds;
+        
+        
+        % Construct a questdlg to ask the user if they are happy with their
+         % well selection
+        choice = questdlg('Are you happy with your well selections?', ...
+        'Well selections for array boundaries', ...
+        'Yes','No','Yes');
+        
+        % Handle response
+        switch choice
+            
+            case 'Yes';
+                disp([choice 'Great, let''s keep going then!'])
+            
+            case 'No';
+                disp([choice 'That''s okay, try again!'])
+                test = 1;
+        end
+     
+    % check whether the user selected array boundaries are correct    
+    if (x_rightwell<x_leftwell || y_bottomwell<y_topwell)        
+        test = 1;
+        
+        title('Oh no! We detected you selected the wells in the wrong order. Please try again. Press any key to continue')
+        pause()
+    else
+        test = 0;
+    end
+  end
+    transform=1;
+end
+
+
+% Determine number of wells per row
+wells_per_row = round((array_bounds(2,1) - array_bounds(1,1)) / horzspacing)+1;
+struct.wells_per_row = wells_per_row;
+
+% Determine number of rows
+rows = round((array_bounds(4,2)-array_bounds(3,2)) / vertspacing)+1;
+
+struct.rows = rows;
```

Comment:
When I resolved the merge conflict, I kept these lines from the roiGenerationEdit branch